### PR TITLE
Update sir-trevor widgets to use IIIF-backed images

### LINKED ISSUE #1673: Update image based SirTrevor widgets to use IIIF
Currently the sir-trevor widgets that use images from documents in the exhibit choose from 3 image derivatives stored as URLs in solr document fields (square, thumb, and full).

We should instead use the manifest stored in the document and IIIF identifiers (as well as cached image API references) similar to how Mastheads and Thumbnails are handled.

This will allow us to change image sizes for widgets to be more performant (as well as adds the potential for more responsiveness).

### PR DESCRIPTION
**Pull request by cbeer**
Fixes #1673 
Fixes https://github.com/sul-dlss/exhibits/issues/315

- [x] consider whether we should support non-IIIF backed image resources, especially for thumbnail-only applications
- [x] add non-IIIF alert (unnecessary. we're falling back to single-image objects with indexed data for non-IIIF stuff)
- [x] use shared IIIF manifest parsing
- [x] refactor away classes supporting non-IIIF-backed image versions
- [x] update uploaded resources to render using IIIF manifests (#1696)
- [x] #1685

### CODE REVIEW

File: app/assets/javascripts/spotlight/blocks/solr_documents_block.js [hunk 1]
```diff
@@ -49,20 +49,55 @@ SirTrevor.Blocks.SolrDocuments = (function(){
...
+      $(this.inner).find('[name$="[iiif_canvas_id]"]').val(data.canvasId);
+    },
     afterPanelRender: function(data, panel) {
       var context = this;
+      var manifestUrl = data.iiif_manifest || data.iiif_manifest_url;
 
-      if (_.isUndefined(data['image_versions'])) {
-        $.getJSON(this.autocomplete_url().replace("%QUERY", "id:" + data.id), function(data) {
-          var doc = context.transform_autocomplete_results(data)[0];
+      $.ajax(manifestUrl).success(
```
Comment (cbeer) [inline app/assets/javascripts/spotlight/blocks/solr_documents_block.js]: This manifest parsing is almost entirely copied from https://github.com/projectblacklight/spotlight/blob/master/app/assets/javascripts/spotlight/search_typeahead.js#L68

**Comment by jkeck** [comment]
@cbeer: I'm getting a bunch of JS errors from the blocks on the edit page (and I believe that it's preventing the blocks icons from rendering)

sir-trevor-blocks

**Comment by cbeer** [comment]
Huh, that's strange. Apparently `default` is a reserved word in js?

**Comment by jkeck** [comment]
I think it is w/ es6 w/ the `export default class Foo`.

**Comment by cbeer** [comment]
I'll take a look next week -- in checking this out, I think I found another bug too.

**Comment by jkeck** [comment]
👍 totally.  All-in-all this is looking great so far though.  Thanks.

**Comment by jkeck** [comment]
@cbeer I'm getting some of the hidden fields set to `undefined` when rendering the edit screen for a block that has already been created.  I believe `undefined` is being persisted as a value in the db and then it is trying to be used to generate image URLs.

undefined

**Comment by cbeer** [comment]
Thanks, I think I've addressed that in 26b3e35.

**Pull request merged**